pipeline/seed_data.py

```diff
@@ -0,0 +1,133 @@
+    connection = engine.connect()
+    return connection
+
+def read_csv(file_path: str) -> list[dict]:
+    """Extracts the relevant information about botanists from the plant data,
+    Appends to a list to seed the 'botanist' table."""
+    with open(file_path, 'r', encoding='utf-8') as file:
+        csv_reader = csv.DictReader(file)
+        return list(csv_reader)
+
+
+def extract_botanist_data(csv_data: list[dict]) -> list[dict]:
+    """Takes the .csv data and returns relevant botanist information as a list of unique dicts."""
+    botanists = []
+    for plant in csv_data:
+        name = plant['Botanist Name']
+        email = plant['Botanist Email']
+        telephone_number = plant['Botanist Phone']
+        botanists.append({"name": name, "email": email, "telephone_number": telephone_number})
+
+    botanist_sets = set([tuple(botanist.items()) for botanist in botanists])
+    unique_botanists = [dict(i) for i in botanist_sets]
+    return unique_botanists
+
+
+def extract_location_data(csv_data: list[dict]) -> list[dict]:
+    """Takes the cleaned .csv data, returns the location 
+    information for each plant as a list of unique dicts."""
+    locations = []
+    for plant in csv_data:
+        continent = plant['Continent']
+        country = standardize_country_name(plant["Country's Initials"])
+        region = plant['Region']
+        locations.append({"region": region, "country": country, "continent": continent})
+
+    location_sets = set([tuple(location.items()) for location in locations])
+    unique_locations = [dict(i) for i in location_sets]
+    return unique_locations
+
+
+def extract_plant_data(csv_data: list[dict],botanist_data: list[dict], location_data: list[dict]) -> list[dict]:
+    """Extracts relevant plant data from the cleaned .csv file, combines with the 
+    botanist and location data, returns a list of unique dicts to seed the plant table."""
+    plants = []
+    for plant in csv_data:
+        plant_id = plant["Id"]
+        name = plant['Name']
+        for botanist in botanist_data:
+            if plant['Botanist Name']== botanist['name']:
+                botanist_num = botanist['telephone_number']
+        for location in location_data:
+            if plant['Region'] == location['region']:
+                location_reg = location['region']
+
+        plants.append({"plant_id": plant_id, "name": name, "botanist_num": botanist_num, "location_reg": location_reg})
+    return plants
+
+
+def seed_botanist_table(conn: Connection, botanist_data: list[dict]) -> None:
+    """Seed the botanist table with the botanist data list."""
+    for botanist in botanist_data:
+        query = sql.text("INSERT INTO s_delta.botanist (name, email, telephone_number) VALUES (:n, :e, :t)")
+        args = ({"n":botanist["name"], "e":botanist["email"], "t":botanist["telephone_number"]})
+        conn.execute(query,args)
```

Comment:
The variables in the dict could be more expanded like name instead of n. However, I understand why its kept short